### PR TITLE
WIP hmac support

### PR DESCRIPTION
**Pull request by southpolesteve**
This still needs work but wanted to open for any feedback on the API:

``` js
import { hmac } from "crypto-hash"

hmac.sha256("my secret key", "text to sign")
hmac.sha256("my secret key", "text to sign", { output: "buffer" })
hmac.sha256("my secret key", "text to sign", { output: "base64" })
```

Not sure if the `hmac` container object makes sense. Could also put them the top-level: 

`import { hmacSha256 } from "crypto-hash" `

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -45,6 +137,9 @@ test('parallel execution', async t => {
 	t.is(results.length, promises.length);
 
 	for (const result of results) {
-		t.is(result, '7d9e515c59bd15d0692f9bc0c68f50f82b62a99bef4b8dc490cec165296210dff005529a4cb84a655eee6ddec82339e6bdbab21bdb287b71a543a56cfab53905');
+		t.is(
+			result,
+			'7d9e515c59bd15d0692f9bc0c68f50f82b62a99bef4b8dc490cec165296210dff005529a4cb84a655eee6ddec82339e6bdbab21bdb287b71a543a56cfab53905'
+		);
```
Comment (sindresorhus) [inline test.js]: Don't do unrelated changes.

**Comment by southpolesteve** [comment]
@sindresorhus Gonna close this for now and revisit later. Still having some internal debate about how to approach this. In general, having syntax that doesn't work on Edge is problematic for us so we may just fork https://github.com/sindresorhus/crypto-hash/issues/13
